app/workers/mail_chimp_worker.rb

```diff
@@ -0,0 +1,12 @@
+class MailChimpWorker
+  include Sidekiq::Worker
+  include Mailchimp
+
+  def perform(user_id)
+    user= User.find(user_id)
+    subscribe_to_mailchimp(ENV['USERS_LIST_ID'], user)
+    if User.subscribe_newsletter
```

Comment:
`subscribe_newsletter` is an attribute of a `User` instance, not a `User` class. So this won't work. You have to call this from from `user` instance. Also since we are not persisting the `subscribe_news_letter`  attribute. You won't get the value from database. Either you have to pass `user` object to the worker instead of `user_id`. Or pass it as an extra parameter to the worker